### PR TITLE
Migrate Redis to indy plugin

### PR DESCRIPTION
**Pull request by felixbarny**
## Checklist

Relates to #1337 

- [x] This is something else
  - [x] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/master/CHANGELOG.asciidoc)

### CODE REVIEW

File: apm-agent-plugins/apm-redis-plugin/apm-jedis-plugin/src/main/java/co/elastic/apm/agent/jedis/JedisInstrumentation.java [hunk 1]
```diff
@@ -41,21 +41,23 @@
...
                 .withPort(thiz.getClient().getPort());
         }
+        return span;
     }
 
-    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class)
-    private static void afterSendCommand(@Nullable @Advice.Local("span") Span span,
+    @Advice.OnMethodExit(onThrowable = Throwable.class, suppress = Throwable.class, inline = false)
+    public static void afterSendCommand(@Nullable @Advice.Enter Object spanObj,
                                          @Nullable @Advice.Thrown Throwable thrown) {
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-redis-plugin/apm-jedis-plugin/src/main/java/co/elastic/apm/agent/jedis/JedisInstrumentation.java:59]: [very minor] fix indent

File: apm-agent-plugins/apm-redis-plugin/apm-lettuce-plugin/src/main/java/co/elastic/apm/agent/lettuce/Lettuce5StopSpanInstrumentation.java [hunk 2]
```diff
@@ -68,17 +66,12 @@
...
-
     public static class OnComplete extends Lettuce5StopSpanInstrumentation {
 
-        @Advice.OnMethodEnter(suppress = Throwable.class)
-        private static void beforeComplete(@Advice.This RedisCommand command) {
+        @Advice.OnMethodEnter(suppress = Throwable.class, inline = false)
+        public static void beforeComplete(@Advice.This RedisCommand<?, ?, ?> command) {
             if (!command.isDone() && !command.isCancelled()) {
-                Span span = commandToSpan.remove(command);
+                Span span = Lettuce5StartSpanInstrumentation.commandToSpan.remove(command);
```
Comment (SylvainJuge) [inline apm-agent-plugins/apm-redis-plugin/apm-lettuce-plugin/src/main/java/co/elastic/apm/agent/lettuce/Lettuce5StopSpanInstrumentation.java:74]: [question] does the import static allows to remove the extra class qualifier ?

**Review by SylvainJuge — APPROVED** [review]
We only have a single class in the `redis-common` module, in the `agent.redis` package, which means it will be loaded in a dedicated plugin classloader. Given the class is stateless that's not really an issue here, but maybe annotating it with `@GlobalState` would avoid this by loading it in the main agent CL.

File: apm-agent-plugins/apm-redis-plugin/apm-jedis-plugin/src/main/java/co/elastic/apm/agent/jedis/JedisInstrumentation.java (hunk 1 above)
Comment (felixbarny) [inline apm-agent-plugins/apm-redis-plugin/apm-jedis-plugin/src/main/java/co/elastic/apm/agent/jedis/JedisInstrumentation.java:59]:
```suggestion
                                        @Nullable @Advice.Thrown Throwable thrown) {
```

**Pull request merged**